sympy/stats/joint_rv.py

```diff
@@ -320,7 +315,9 @@ def symbols(self):
     def pdf(self, *x):
```

Comment:
Can this be rewritten as
```python
def pdf(self, *x):
    expr, rvs = self.args[0], self.args[1]
    marginalise_out = [i for i in random_symbols(expr) if i not in rvs]
    if isinstance(expr, CompoundDistribution):
        syms = Dummy('x', real=True)
        expr = expr.args[0].pdf(syms)
    elif isinstance(expr, JointDistribution):
        count = len(expr.domain.args)
        x = Dummy('x', real=True, finite=True)
        syms = [Indexed(x, i) for i in count]
        expr = expr.pdf(syms)
    else:
        syms = [rv.pspace.symbol if isinstance(rv, RandomSymbol) else rv.args[0] for rv in rvs]
    return Lambda(syms, self.compute_pdf(expr, marginalise_out))(*x)
```
`random_symbols` already identifies the atom of an Indexed's name:
```python
>>> r=RandomSymbol(Symbol('r'))
>>> q=IndexedBase(r)[i]
>>> q.atoms()
{i, r}
>>> random_symbols(q)
[r]
```
So adding on to `marginalise_out` symbols that are already present will just create a duplicate in the list.